src/components/Swap/SwapActionButton/AllowanceButton.tsx

```diff
@@ -47,28 +50,39 @@ export default function AllowanceButton({ token, isApprovalLoading, approveAndPe
         icon: Spinner,
         message: pendingApproval ? (
           <EtherscanLink type={ExplorerDataType.TRANSACTION} data={pendingApproval}>
-            <Trans>Approval pending</Trans>
+            <Trans>{token?.symbol ?? 'Token'} approval pending</Trans>
           </EtherscanLink>
         ) : (
-          <Trans>Approval pending</Trans>
+          <Trans>{token?.symbol ?? 'Token'} approval pending</Trans>
         ),
       }
     } else if (isFailed) {
       return {
-        message: t`Approval failed`,
+        message: t`${token?.symbol ?? 'Token'} approval failed`,
         onClick,
+        color: 'warning',
       }
     } else {
       return {
-        tooltipContent: t`Permission is required for Uniswap to swap each token. This will expire after one month for your security.`,
-        message: t`Approve use of ${token?.symbol ?? 'token'}`,
+        tooltipContent: t`Permission is required for Uniswap smart contract to use each token. This will expire after one month for your security.`,
```

Comment:
Copy needs work: it should either be "for Uniswap to" or "for **the** Uniswap smart contract to". As-is is grammatically off.

Reply:
i'll add `the` - copied this directly from the figma